pkg/http/api_test.go

```diff
@@ -2817,6 +2817,23 @@ func TestV1Beta1Workflow(t *testing.T) {
 		assert.Nil(t, resp.ErrorBody)
 	})
 
+	t.Run("Terminate with recursive true", func(t *testing.T) {
+		// Note that in case of recursive termination, MockWorkflow intentionally returns fake error, even when it is not an actual error.
+		// This is to test that recursive flag is being passed correctly to the workflow component.
+
+		apiPath := "v1.0-beta1/workflows/dapr/instanceID/terminate"
+		data := struct {
+			Recursive bool `json:"recursive"`
+		}{
+			Recursive: true,
+		}
+		body, _ := json.Marshal(data)
+		resp := fakeServer.DoRequest("POST", apiPath, body, nil)
+		assert.Equal(t, 500, resp.StatusCode)
+		assert.Equal(t, "ERR_TERMINATE_WORKFLOW", resp.ErrorBody["errorCode"])
+		assert.Equal(t, fmt.Sprintf(messages.ErrTerminateWorkflow.Message(), "instanceID", daprt.ErrFakeWorkflowTerminateRecursiveError), resp.ErrorBody["message"])
+	})
```

Comment:
can we have this as a query string parameter instead?

Reply:
Done